src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/PerformanceTrackerService.cs

```diff
@@ -58,24 +57,32 @@ internal PerformanceTrackerService(double minLOFValue, double averageThreshold,
             _stddevThreshold = stddevThreshold;
```

Comment:
@jmarolf I don't understand how these 3 values here (`minLOFValue`, `averageThreshold`, `stddevThreshold`) affect the reporting. Do we need different values for background document analysis and span analysis based on use of different max/min sample sizes for both the performance queues? Honestly, I don't have any clue here.

Reply:
These are used here: https://github.com/dotnet/roslyn/blob/main/src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/PerformanceTrackerService.cs#L171-L228

minOFValue: minimum outlier factor value
averageThreshold: average threshold
stddevThreshold: standard deviation threshold

the two "threshold" values are used as our first filter. This means we exclude results where analysis happens "too fast". We will want to change `averageThreshold` as span analysis is (presumably) going to be much faster. 

minOFValue is how we identify something as an "outlier" according to "local outlier factor" analysis. The reality is that we will likely need to get a sampling of this data first in order to determine what this value should be for span diagnostics but presumably it will not be the same in the span case.